### PR TITLE
Part5 and Part6

### PR DESCRIPTION
**Pull request by alexey-1991**
### part 5
image
### part 6
image

### [Demo version](https://reactjs2019spa.herokuapp.com/)

### CODE REVIEW

File: config/webpack/webpack.production.config.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 const path = require('path');
 const merge = require('webpack-merge');
 const MiniCssExtractPlugin = require('mini-css-extract-plugin');
-const UglifyJsPlugin = require('uglifyjs-webpack-plugin');
+const TerserPlugin = require('terser-webpack-plugin');
```
Comment (shubich) [inline config/webpack/webpack.production.config.js]: 👍

File: package.json [hunk 2]
```diff
@@ -78,6 +78,7 @@
   "dependencies": {
     "autoprefixer": "^9.4.3",
     "axios": ">=0.18.0",
+    "babel-polyfill": "^6.26.0",
```
Comment (shubich) [inline package.json]: how about **@babel/polyfill**?

File: src/components/FilmCounter/FilmCounter.js [hunk 3]
```diff
@@ -1,9 +1,29 @@
...
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
 
 import styles from './FilmCounter.scss';
 
-const FilmCounter = () => (
-  <p className={styles.text}>7 films found </p>
-);
+const FilmCounter = ({ total }) => {
+  const formatNumber = (number) => {
```
Comment (shubich) [inline src/components/FilmCounter/FilmCounter.js]:
```javascript
const formatNumber = useCallback((number) => { ... }, [dependencies])
```

File: src/components/FilmItem/FilmItem.js [hunk 4]
```diff
@@ -2,36 +2,43 @@ import React from 'react';
...
-    <div className={styles.title}>{title.toUpperCase()}</div>
-    <div className={styles.genre}>
-      {genre.map(item => (
-        <p key={item} className={styles.genreItem}>{item}</p>
-      ))}
+  release_date: date,
+  poster_path: image,
+  genres,
+}) => {
+  const year = (new Date(date)).getFullYear();
```
Comment (shubich) [inline src/components/FilmItem/FilmItem.js]: useMemo

File: src/components/FilmItem/FilmItem.js [hunk 5]
```diff
@@ -2,36 +2,43 @@ import React from 'react';
...
-  genre: PropTypes.arrayOf(
-    PropTypes.string,
-  ).isRequired,
+  release_date: PropTypes.string.isRequired,
+  poster_path: PropTypes.string,
+  genres: PropTypes.arrayOf(PropTypes.string).isRequired,
 };
 
-export default FilmList;
+export default FilmItem;
```
Comment (shubich) [inline src/components/FilmItem/FilmItem.js]: export default React.memo(FilmItem);

File: src/components/FilmList/__tests__/FilmList.test.js [hunk 6]
```diff
@@ -1,46 +1,99 @@
...
+      },
+    };
+  });
+
+  test('render [defaultProps]', () => {
+    store.collection.data = [];
+    const myStore = mockStore(store);
+    const component = mount(
+      <Provider store={myStore}>
+        <FilmList history={myHistory} />
```
Comment (shubich) [inline src/components/FilmList/__tests__/FilmList.test.js]: try to separate component from container and than you probably will not need to mock store

File: src/components/WatchLocation/WatchLocation.js [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+      getCollection(url);
+      return;
+    }
+
+    if (path === paths.preview) {
+      const { id } = params;
+      const url = getUrlFilmById(id);
+      getFilm(url);
+    }
+  }, [currentLocation]);
```
Comment (shubich) [inline src/components/WatchLocation/WatchLocation.js]: do you have [eslint-plugin-react-hooks](https://www.npmjs.com/package/eslint-plugin-react-hooks)?
all dependencies must be in []

File: src/components/WatchLocation/WatchLocation.js [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+import paths from 'config/paths';
+import {
+  getUrlFilms,
+  filmByIdApi as getUrlFilmById,
+  searchType,
+  sortType,
+  sortOrderType,
+} from 'utils/api';
+
+const WatchLocation = ({
```
Comment (shubich) [inline src/components/WatchLocation/WatchLocation.js]: first time see approach like this 🙂

File: src/modules/collection/reducer.js [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    case START_LOADING_COLLECTION:
+      return {
+        ...state, data: [], loading: true, error: null,
+      };
+    case STOP_LOADING_COLLECTION:
+      return { ...state, loading: false };
+    case UPDATE_COLLECTION:
+      return { ...state, data: action.payload };
+    case SET_ERROR_COLLECTION:
+      return { ...state, error: action.payload };
```
Comment (shubich) [inline src/modules/collection/reducer.js]: reset loading?

Comment (alexey-1991) [reply src/modules/collection/reducer.js]: there is a separate action to stop loading

File: package.json (hunk 2 above)
Comment (alexey-1991) [reply package.json]: fixed

File: src/components/FilmCounter/FilmCounter.js (hunk 3 above)
Comment (alexey-1991) [reply src/components/FilmCounter/FilmCounter.js]: fixed

File: src/components/FilmItem/FilmItem.js (hunk 4 above)
Comment (alexey-1991) [reply src/components/FilmItem/FilmItem.js]: fixed

File: src/components/FilmItem/FilmItem.js (hunk 5 above)
Comment (alexey-1991) [reply src/components/FilmItem/FilmItem.js]: fixed

File: src/components/WatchLocation/WatchLocation.js (hunk 7 above)
Comment (alexey-1991) [reply src/components/WatchLocation/WatchLocation.js]: fixed

**Pull request merged**